elastic/kibana
# [ftr] enable mock-idp-plugin for stateful (deployment-agnostic) tests

## Summary

closes #190221

This PR enables `mock-idp-plugin` when Kibana is started with stateful FTR config for deployment-agnostic tests:

```
 node scripts/functional_tests_server --config=x-pack/test/api_integration/deployment_agnostic/configs/stateful/platform.stateful.config.ts
```

<img width="1574" alt="image" src="https://github.com/user-attachments/assets/494e89ee-cd65-4dde-86da-a5e2c28ec40d">

You can pick up one of the the role defined for stateful SAML authentication in https://github.com/elastic/kibana/blob/main/packages/kbn-es/src/stateful_resources/roles.yml 

Note: this plugin is only enabled locally for a better manual testing experience, it is **not loaded on CI**

It is done to unify DevEx when folks work on deployment-agnostic tests and would like to confirm the functionality under the same role for both stateful and serverless deployments.

Thanks @azasypkin for the help, again :)

How to test: 
- start the servers using  `x-pack/test/api_integration/deployment_agnostic/configs/stateful/platform.stateful.config.ts` config and go to `http://localhost:5620`
- try to login with different roles, make sure valid role is applied in top right profile menu
